### PR TITLE
#258 avoiding writing useless informations in par and dyd when a comp…

### LINKED ISSUE #258: Avoiding writing of useless informations in par and dyd when an object does not exists
In case a component it is not present in the input file (e.g. SvarCs, loads, ...) no information should be written in dynawo input files

### PR DESCRIPTION
**Pull request by marcochiaramello**
…onent definition does not exists
Closes #258

### CODE REVIEW

File: sources/Outputs/include/Dyd.h [hunk 1]
```diff
@@ -191,27 +191,33 @@ class Dyd {
    *
    * Create Model signal N model
    *
+   * @param def the dyd definition
+   *
    * @returns black box models
    */
-  static std::vector<boost::shared_ptr<dynamicdata::BlackBoxModel>> writeConstantsModel();
+  static std::vector<boost::shared_ptr<dynamicdata::BlackBoxModel>> writeConstantsModel(const DydDefinition& def_);
```
Comment (rosiereflo) [inline sources/Outputs/include/Dyd.h:198]: no "_" in parameter names

File: tests/outputs/TestPar.cpp [hunk 2]
```diff
@@ -244,7 +244,7 @@ TEST(TestPar, writeStaticVarCompensator) {
   outputPath.append(filename);
   dfl::inputs::Configuration::ActivePowerCompensation activePowerCompensation(dfl::inputs::Configuration::ActivePowerCompensation::P);
   dfl::outputs::Par parWriter(dfl::outputs::Par::ParDefinition(basename, dirname, outputPath.generic_string(), {}, {}, activePowerCompensation, {},
-                                                               manager, {}, {}, {}, svarcs));
+                                                               manager, {}, {}, {}, svarcs, {}));
```
Comment (rosiereflo) [inline tests/outputs/TestPar.cpp:247]: Maybe adding a test with no load to test that the connection is not dumped? (same goes for dyd)

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: sources/Outputs/include/Dyd.h (hunk 1 above)
Comment (marcochiaramello) [reply sources/Outputs/include/Dyd.h:198]: ok, thx

File: tests/outputs/TestPar.cpp (hunk 2 above)
Comment (marcochiaramello) [reply tests/outputs/TestPar.cpp:247]: I'd rather add a test on loads as missing, for the other tests loads are already missing

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**